R/stack.R

```diff
@@ -582,30 +583,64 @@ getDotVars <- function(env) {
 #' 
 #' Gets a representation of a promise without forcing the promise
 #' 
+#' @aliases .vsc.promise
 #' @param name The name of the variable
 #' @param env The environment in which to evaluate
-#' @return A named list containing the expression that will be evaluated, the environment in which it will be evaluated and a string representation of the expression
+#' @return An object of class \code{".vsc.promise"}; a named list containing the expression that will be evaluated, the status whether the promise has already been evaluated, the value if it has already been evaluated, and the environment in which the unevaluated promise will be evaluated. 
+#' @examples 
+#' ## create a promise
+#' e <- new.env()
+#' delayedAssign("x", {message("evaluating promise..."); 1L}, assign.env = e)
+#' vscDebugger:::getPromiseVar("x", e)
+#' 
+#' ## evaluate it...
+#' print(e$x)
+#' 
+#' ## is it still a promise? (It depends...)
+#' stopifnot(vscDebugger:::isPromise("x", e, strict = FALSE))
+#' stopifnot(!vscDebugger:::isPromise("x", e, strict = TRUE))
+#' 
+#' ## get info again
+#' vscDebugger:::getPromiseVar("x", e)
 #' 
 getPromiseVar <- function(name, env) {
-  promiseExpr <- pryr:::promise_code(name, env)
-  promiseCode <- try(paste0(toString(promiseExpr), collapse = ';'), silent=getOption('vsc.trySilent', default=TRUE))
-  if (class(promiseCode) == 'try-error') promiseCode <- '???'
-  promiseEnv <- pryr:::promise_env(name, env)
-  var <- list(
-    promiseCode = promiseCode,
-    promiseEnv = promiseEnv,
-    promiseExpr = promiseExpr
+  structure(
+    getPromiseInfo(name, env),
+    class = c(".vsc.promise", ".vsc.InternalClass"
```

Comment:
This line seems to be missing a `)`:
`    class = c(".vsc.promise", ".vsc.InternalClass")`

Reply:
Fixed